hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/fsck/MissingContainerTask.java

```diff
@@ -0,0 +1,152 @@
+import org.hadoop.ozone.recon.schema.tables.pojos.ReconTaskStatus;
+import org.jooq.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * Class that scans the list of containers and keeps track of containers with
+ * no replicas in a SQL table.
+ */
+public class MissingContainerTask {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(MissingContainerTask.class);
+
+  private OzoneStorageContainerManager scm;
+  private MissingContainersDao missingContainersDao;
+  private ReconTaskStatusDao reconTaskStatusDao;
+  private static final long INTERVAL = 5 * 60 * 1000L;
+
+  private Thread fsckMonitor;
+  private volatile boolean running;
+
+  public MissingContainerTask(ReconStorageContainerManagerFacade scm,
+                              Configuration sqlConfiguration) {
+    this.scm = scm;
+    this.missingContainersDao = new MissingContainersDao(sqlConfiguration);
+    this.reconTaskStatusDao = new ReconTaskStatusDao(sqlConfiguration);
+    String taskName = getClass().getSimpleName();
+    ReconTaskStatus reconTaskStatusRecord = new ReconTaskStatus(
+        taskName, 0L, 0L);
+    if (!reconTaskStatusDao.existsById(taskName)) {
+      reconTaskStatusDao.insert(reconTaskStatusRecord);
+      LOG.info("Registered {} task ", taskName);
+    }
+  }
+
+  public void start() {
+    if (!isRunning()) {
+      LOG.info("Starting Missing Container Monitor Thread.");
+      running = true;
+      fsckMonitor = new Thread(this::run);
+      fsckMonitor.setName("MissingContainerMonitor");
+      fsckMonitor.setDaemon(true);
+      fsckMonitor.start();
+    } else {
+      LOG.info("Missing Container Monitor Thread is already running.");
+    }
+  }
+
+  /**
+   * Stops Replication Monitor thread.
+   */
+  public synchronized void stop() {
+    if (running) {
+      LOG.info("Stopping Missing Container Monitor Thread.");
+      running = false;
+      notifyAll();
+    } else {
+      LOG.info("Missing Container Monitor Thread is not running.");
+    }
+  }
+
+  public synchronized void run() {
+    try {
+      while (running) {
+        long start = Time.monotonicNow();
+        long currentTime = System.currentTimeMillis();
+        ContainerManager containerManager = scm.getContainerManager();
+        final Set<ContainerID> containerIds =
+            containerManager.getContainerIDs();
+        containerManager.getContainerIDs().forEach(containerID ->
+            processContainer(containerID, currentTime));
+
+        LOG.info("Missing Container Monitor Thread took {} milliseconds for" +
+                " processing {} containers.", Time.monotonicNow() - start,
+            containerIds.size());
+        wait(INTERVAL);
+      }
+    } catch (Throwable t) {
+      // When we get runtime exception, we should terminate SCM.
+      LOG.error("Exception in Replication Monitor Thread.", t);
```

Comment:
Is that necessarily true for Recon, could there be a benign problem that we could recover from?

Reply:
We dont need to terminate Recon. Will remove it.